src/tools/generic_key.rs

```diff
@@ -0,0 +1,56 @@
+use super::prelude::*;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct GenericKey<const N: usize>([u8; N]);
```

Comment:
We now have `ArrayKey<>` struct, so this `GenericKey<>` should be removed and `ArrayKey<>` should be used instead